src/mbgl/geometry/feature_index.cpp

```diff
@@ -194,4 +196,46 @@ void FeatureIndex::setBucketLayerIDs(const std::string& bucketLeaderID, const st
     bucketLayerIDs[bucketLeaderID] = layerIDs;
 }
 
+static LatLng screenCoordinateToLatLng(const ScreenCoordinate& p,
+                                       const TransformState& s,
+                                       LatLng::WrapMode wrapMode = LatLng::Wrapped) {
+    ScreenCoordinate flippedPoint = p;
+    flippedPoint.y = s.getSize().height - flippedPoint.y;
+    return s.screenCoordinateToLatLng(flippedPoint, wrapMode);
+}
+static Point<double> project(const LatLng& c, const TransformState& s) {
+    LatLng unwrappedLatLng = c.wrapped();
+    unwrappedLatLng.unwrapForShortestPath(s.getLatLng(LatLng::Wrapped));
+    return Projection::project(unwrappedLatLng, s.getScale());
+}
+
+void FeatureIndexDynamic::query(std::unordered_map<std::string, std::vector<Feature>>& result,
+                                const mbgl::ScreenLineString& queryGeometry,
+                                const TransformState& state) const
+{
+    // Convert queryGeometry to mercator
+    auto p = queryGeometry.front();
+    LatLng c = screenCoordinateToLatLng(p, state);
+    Point<double> pm = project(c, state);
+    Point<int64_t> coord = {int64_t(pm.x), int64_t(pm.y) };
+    for (const auto &f: features) {
+        // hit testing
+        bool hit = pointWithinPolygon(coord, *f.envelope);
+        if (hit) {
```

Comment:
`if (pointWithinPolygon(coord, *f.envelope))`

Reply:
i left it as it seems clearer, this way (and easier to debug)